app/api/constituency_lookup/[postcode]/[[...address]]/route.ts

```diff
@@ -84,16 +135,54 @@ export async function GET(
     };
     return NextResponse.json(response);
   } else {
-    // TODO: Use DemocracyClub API to lookup postcode and populate the addresses array, so users can select their
-    // specific address, rather than us expecting to know (or find out) their constituency.
     console.log(
       `Multiple constituencies found for postcode ${normalizedPostcode}`,
     );
+
     const response: ConstituencyLookupResponse = {
       postcode: params.postcode,
       addressSlug: params.address?.[0],
       constituencies: constituencies,
     };
+
+    //see if DC api will give us anything useful
+    //TODO might be worth setting a timeout on this?
+    //TODO remove this if statement, (here so we can see DC API failure!)
+    //let dc_data = null;
+    //if (normalizedPostcode != "DE30GU")
+    // dc_data = await fetch_dc_api(normalizedPostcode, params.address?.[0]);
+
+    const dc_data = await fetch_dc_api(normalizedPostcode, params.address?.[0]);
+
+    // See if DC data can return a more useful response.
+    // Possibly it just matches 1 constituency in which case return that
+    // Possibly it gives us an address picker which we can return.
+    if (dc_data?.boundary_changes) {
+      //Democracy Club data shows this postcode has a single constituency
+      const gss =
+        dc_data.boundary_changes.new_constituencies_official_identifier.substring(
+          4,
+        );
+      const dc_constituency = constituencies.filter((constituency) => {
```

Comment:
Might be worth renaming `constituencies` above to `db_constituencies` or something, just to avoid confusion now that we have data coming from 2 possible sources :sweat_smile: